### ISSUE TITLE
Allow to use local fonts in texts

### ISSUE THREAD

**Issue by lakejason0** [new-sankaku/stable-diffusion-webui-simple-manga-maker#52]
Fonts like Klee One have got several derivative fonts for extending character set for Chinese users, like [LXGW WenKai](https://github.com/lxgw/LxgwWenkai), their Traditional Chinese variant is also now on Google fonts.

However, I don't think we should always ask in this repo to "add this font add that font etc. etc.", maybe just let user to specify a font family name; if it's installed in the computer, then it can be used; if it's not, it looks like there's no change anyway. I manually edited the font dropdown menu to add LXGW WenKai, and it works just fine.

Though personally I would ask for advanced OpenType feature toggles (like `halt` for punctuations and style sets provided by fonts), but that's just me, it's kinda beyond the "simple". For now, just allow the user to manually enter an font-family name would be enough.

**Comment by new-sankaku** [comment]
I understand.
It looks like it will support local fonts, web fonts (css), and font files. I think it will be done by the weekend.

**Comment by new-sankaku** [comment]
I've enabled registration so please give it a try.

![image](https://github.com/user-attachments/assets/4b41194c-60e6-465e-9dbf-204655b3284e)
![image](https://github.com/user-attachments/assets/1ee41639-51fb-444a-b36e-391f5a66881f)

**Comment by lakejason0** [comment]
kinda trying, but I can no longer add texts somehow?

**Comment by lakejason0** [comment]
It only supports the English font name, but fonts would actually have localized name in their font file (and can be recognized by the browser). However, when I input the localized (like Chinese) font name, the console would say Network Error?

**Comment by new-sankaku** [comment]
Is your current situation as follows?
 Tried to add fonts using "添加字体"
 Added Chinese font names using "添加本地字体（每行一个）"
 Network errors appear in the console

I am a Japanese user, so I tested using Japanese font names. I have confirmed that the following font names from Excel's font list were added as local fonts and properly reflected in the text (both horizontal and vertical writing).
[These are font names combining Japanese katakana, kanji, and English]
"メイリオ"
"游ゴシック"
"BIZ UDPゴシック"
"HG丸ｺﾞｼｯｸM-PRO"
"HGP創英角ﾎﾟｯﾌﾟ体"
"UD デジタル 教科書体 NK-B"
"ＭＳ Ｐゴシック"

While probably unrelated, I recently fixed an issue where newly added fonts weren't displaying in real-time in the font selection dropdown.
Your situation might be related to character encoding. Could you tell me which font names you tried to add?

**Comment by lakejason0** [comment]
I think the problem is fixed after I updated to the latest commit... Idk why that is happening.

**Comment by new-sankaku** [comment]
Oh, I'm glad it's fixed! Enjoy your day!

**Comment by new-sankaku** [comment]
Oh... Let's investigate.

**Comment by new-sankaku** [comment]
Hmm, I looked into it, but I'm not sure... 😢
If you're on Windows, there should be a list of fonts here, but does it match the family name you specified?

C:\Windows\Fonts

I added some pattern processing just to be sure. But I don't think this will solve the problem.

**Comment by lakejason0** [comment]
Of course it matches. If I search the English name in Fonts, it would show "No result".

**Comment by new-sankaku** [comment]
By the way, how did you try it, lakejason0?

**Comment by lakejason0** [comment]
...and nope, I updated to the latest commit `6bcb384` and it still shows Network Error. Maybe add a hint to the interface that "users should try English names first" or sth?

**Comment by lakejason0** [comment]
> By the way, how did you try it, lakejason0?

Like I click add font then use the "add local font (one per line)". I enter the fonts name on each line, and some of them would show Network Error.

**Comment by new-sankaku** [comment]
Yes, I can't find the answer even after searching, so I'll just put a help message here! Thank you!
